server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -296,6 +297,19 @@ public boolean hasMappings() {
         return mappingLookup.hasMappings();
     }
 
+    public List<String> nestedMappings() {
+        if (mapperService.documentMapper() == null) {
```

Comment:
would it be possible to not add more accesses to mapperService but rather use MappingLookup? I think it exposes objectMappers too

Reply:
Great, will take a look.

Reply:
@javanna pushed 412bb74, hope that is what you had in minde with this comment

Reply:
yes it is, thanks!